[CS2113-T15-3] SplitLiang

mukund1403 opened the pull request:
Welcome to SplitLiang, the application designed to handle the splitting of bills among groups (equally or unequally).

mukund1403 committed:
fix checkstyle errors in Parser and Expenses class

mukund1403 committed:
fix issues found during code review

mukund1403 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T15-3/tp

Cohii2 committed:
Create UserInterface Class

Create UserInterface Class, MessageType Enum for use with class, and UserInterfaceTest

Cohii2 committed:
Merge pull request #68 from mukund1403/master

Add description and unequal split to Expenses

MonkeScripts committed:
Add test code

MonkeScripts committed:
Fix checkstyle

Cohii2 committed:
Create Pair class

Copied CS2030 utility function Pair class for use with Expense and Balance class.

Cohii2 committed:
Fix checkstyle errors

hafizuddin-a committed:
Merge branch 'master' into hafiz-Database

Cohii2 committed:
Fix IO tests

Commit:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T15-3/tp into hafiz-Database

# Conflicts:
#	src/main/java/seedu/duke/Expense.java

Commit:
Update GroupStorage

Commit:
Merge branch 'hafiz-Database' of https://github.com/hafizuddin-a/tp into hafiz-Database

# Conflicts:
#	src/main/java/seedu/duke/Expense.java

Cohii2 committed:
Edit comments

Commit:
Fix checkstyle

MonkeScripts committed:
Merge pull request #70 from Cohii2/hengJunxiang-pair

Create Pair class

hafizuddin-a committed:
Merge branch 'master' into hafiz-Database

avrilgk committed:
Merge pull request #64 from hafizuddin-a/hafiz-Database

Implement database

Cohii2 committed:
Merge pull request #65 from MonkeScripts/currency

Add currency exchange functionality

mukund1403 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T15-3/tp

MonkeScripts committed:
Gambling feature

mukund1403 committed:
update pair from javafx to inbuilt class and fix issue with equal split

MonkeScripts committed:
Clean out comments

Cohii2 committed:
Add citation

Add citation for Pair class.

Cohii2 committed:
Merge pull request #73 from Cohii2/master

Add citation

MonkeScripts committed:
Clean out checkstyle bugs

MonkeScripts committed:
Fix helpTest

MonkeScripts committed:
Change EXPECTED.TXT for new help

Cohii2 committed:
Update Balance Unit Tests

Cohii2 committed:
Revert "Update Balance Unit Tests"

This reverts commit 7eee70c9a7bd3053cb75bc66b46c147089e14ce3.

Cohii2 committed:
Update Balance Unit Test

Cohii2 committed:
Merge pull request #71 from MonkeScripts/gambling

Gambling

Cohii2 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T15-3/tp

mukund1403 committed:
fix checkstyle error

mukund1403 committed:
Merge pull request #69 from Cohii2/hengJunxiang-ui

Create UserInterface Class

MonkeScripts committed:
Merge pull request #72 from mukund1403/master

update pair from javafx to inbuilt class and fix issue with equal split

MonkeScripts committed:
Merge pull request #75 from Cohii2/master

Update Balance Tests

avrilgk committed:
Settle up function

avrilgk committed:
Fix test fails

avrilgk committed:
Fix test fails

avrilgk committed:
Fix test fails

avrilgk committed:
Updated settle function

mukund1403 committed:
add expense command class, list command class, commands package

mukund1403 committed:
Merge pull request #77 from avrilgk/bugfix-and-settleup

Settle up function

mukund1403 committed:
fix merge conflicts

avrilgk committed:
Updated DG for settle function

Cohii2 committed:
Merge pull request #83 from mukund1403/master

add expense command class, list command class, commands package

hafizuddin-a committed:
Merge pull request #84 from avrilgk/update-settle

Updated DG for settle function

Commit:
Add UniversalExceptions to exceptions package

Commit:
Optional for loadedgroup

Commit:
Update UG and DG

hafizuddin-a committed:
Merge pull request #85 from avrilgk/Avril-UG

Update UG and DG

Commit:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T15-3/tp into hafiz-FixDatabase

Commit:
Update UG

Commit:
Fix checkstyle

Commit:
Add diagram

Cohii2 committed:
Update User Guide

Update Balance of user guide.

Cohii2 committed:
Merge pull request #88 from Cohii2/master

Update User Guide

mukund1403 committed:
add expense feature to user guide

mukund1403 committed:
Merge pull request #89 from mukund1403/master

add expense feature to user guide

Cohii2 committed:
Fix issue with end programme

When user enters 'bye' current group is saved to file.

Cohii2 committed:
Fix checkstyle error

Forgot to delete text from merge conflict.

Commit:
Add GroupNameChecker

Commit:
Fix GroupNameChecker

Commit:
Refactoring of settle to remove arrowhead code and settle JUNIT testing

MonkeScripts committed:
Update UG

Cohii2 committed:
Fix Balance Test

Commit:
Update loadGroupName

Commit:
Fix Checkstyle

mukund1403 committed:
Merge pull request #90 from Cohii2/hengJunxiang-storage

Fix issue with end programme

mukund1403 committed:
add member check to expense class

Cohii2 committed:
Merge pull request #94 from mukund1403/master

add member check to expense class

Cohii2 committed:
Merge branch 'master' into hafiz-MoreDatabase

Cohii2 committed:
Merge pull request #93 from hafizuddin-a/hafiz-MoreDatabase

More Database fix

mukund1403 committed:
Merge pull request #91 from avrilgk/group-edit

Refactoring of settle to remove arrowhead code and settle JUNIT testing

mukund1403 committed:
Merge branch 'master' into hafiz-Guides

mukund1403 committed:
Merge pull request #87 from hafizuddin-a/hafiz-Guides

Edit UG and DG

mukund1403 committed:
Merge pull request #92 from MonkeScripts/gambling

Update UG

mukund1403 committed:
Merge pull request #86 from hafizuddin-a/hafiz-FixDatabase

Partial fix database

jasraa commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,131 @@
+3. Prints a success message indicating that the group has been created.
+4. Adds the new `Group` object to the `groups` list.
+5. Returns the newly created `Group` object.
+6. If the group already exists, prints an error message indicating that the group already exists.
```
Good job on providing detailed steps! It would be helpful to specify how errors are handled in the system. For example, are exceptions thrown, or are error codes returned?

luozihui2003 commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,131 @@
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Help menu feature
```
there could be an architecture diagram of the high-level design of the app, of how the different features interact, to make the overall app design clearer

luozihui2003 commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,131 @@
+5. Returns the newly created `Group` object.
+6. If the group already exists, prints an error message indicating that the group already exists.
+
+### Add Member to Group feature
```
it is not very clear how the user commands are parsed, as there is no information about how the basic components (such as Ui, Model, Logic, Storage) interact with each other before explaining the features

jasraa commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,131 @@
+3. Adds the new `User` object to the `members` list.
+4. Prints a success message indicating that the member has been added to the group.
+
+![Sequence Diagram](AddMember.png)
```
Sequence Diagram is well done and formatted nicely! It would be good to show the "isMember( )" method in the sequence diagram as well :)

luozihui2003 replied:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/110605459/744f0235-d4d7-48e4-8d7b-9678cf065f36)
the alt frame should have an [else] condition

jasraa commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,131 @@
++ `Expenses#payees()` - Returns all the members involved in the transaction
+
+These operations are exposed in the Expense class through the `getPayerName()`, `getTotalAmount()`, and `getPayees()`
+functions respectively.
```
The implementation for this feature is clearly stated and easy to follow. Would it be better to also include a sequence diagram to trace the code?

luozihui2003 commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,131 @@
+4. Prints a success message indicating that the member has been added to the group.
+
+![Sequence Diagram](AddMember.png)
+
```
Activation bar of Group should not extend beyond the [else] part of alt frame, as Group is not being used.

ParthGandhiNUS commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,131 @@
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
 
 ## Design & implementation
```
Perhaps can also consider having class diagrams, coupled with the suggestion for Architecture Diagram, to provide new developers with a more holistic view of the project

ParthGandhiNUS commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,131 @@
+
 ### Target user profile
 
 {Describe the target user profile}
```
Perhaps can consider removing the prompts for the different sections once you guys have started writing more stuff for that particular section.

jasraa commented on docs/DeveloperGuide.md:
```diff
@@ -2,28 +2,131 @@
+The Settle feature is facilitated through the Settle class.
+It allows a user to settle the debts between two users in a Group.
+
+ The `Settle` class contains a `settleDebt(String userName1, String userName2)` method.
```
Nice feature! However, the Settle class does not include a settleDebt( ) method that takes in 2 string inputs. It would be helpful to either include this method in the Settle class or clarify where it is implemented.

Commit:
Add PPP

Commit:
Fix checkstyle

avrilgk committed:
Merge pull request #152 from hafizuddin-a/hafiz-PPP

Add ppp

Commit:
Bye command exit at start

- no longer asks for name at start

Commit:
Update non-existent group error message

- issue #109

avrilgk committed:
Updated PPP

Commit:
Ignore case for member

- issue #126

Commit:
Add isLoading flag

- isLoading flag prevents the console for printing the statments for adding member and expenses when loading file
- resolve issue #119 of printing error message from bye command when not in any group

Commit:
Update isLoading in finally

Commit:
Fix double message group bug

- moved check for whether user is already in group before creating new group to prevent duplicate messages
- modified check to compare current group with group being accessed to avoid false positives
- if user is already in group, return empty Optional to prevent further execution
- issue #123

mukund1403 committed:
Merge pull request #154 from hafizuddin-a/hafiz-PEDDatabase

Fix database issues from PED

hafizuddin-a committed:
Merge pull request #153 from avrilgk/PPP

Updated PPP for avrilgk

avrilgk committed:
fixed exit and enter group bugs

hafizuddin-a committed:
Merge pull request #155 from avrilgk/fixing-group-bugs

fixed exit and enter group bugs

Commit:
Exceptions for tampered file

Commit:
Remove unused import

Commit:
More exceptions

avrilgk committed:
Merge pull request #156 from hafizuddin-a/hafiz-ImproveDatabase

Exceptions for tampered file

Commit:
Fix Loading

Commit:
Add content page and summary table

Commit:
Add more format for member

Commit:
Add delete file

avrilgk committed:
Merge pull request #157 from hafizuddin-a/hafiz-DatabaseOops

Fix Loading

Commit:
Update delete command

- allow user to delete saved groups before loading them

Commit:
Add Javadoc for delete exception

Commit:
Use Universalexceptions

Commit:
Fix checkstyle

mukund1403 committed:
add delete expense feature and fix all issues from pe-d

Commit:
Update DG

- followed addressbook and add more diagrams

Commit:
Add images

avrilgk committed:
Settle update

mukund1403 committed:
Merge pull request #159 from hafizuddin-a/hafiz-UG

Update UG

mukund1403 committed:
add junit testing for ExpenseCommand

MonkeScripts committed:
Merge pull request #161 from mukund1403/master

add delete expense feature and fix all issues from pe-d

avrilgk committed:
update settle function and fixed all PE-D bugs

avrilgk committed:
Merge pull request #160 from hafizuddin-a/hafiz-DeleteGroup

Update delete group

avrilgk committed:
fixed group tests

Commit:
Add diagrams folder

Cohii2 committed:
Separate argument handling into BalanceCommand class

avrilgk committed:
fixed spelling

Cohii2 committed:
Update printing to new UI

avrilgk committed:
fixed spelling

Cohii2 committed:
Add docs to important functions in Balance and BalanceCommand classes

Cohii2 committed:
Fix Bug

Fix bug where member is in group but message says otherwise

MonkeScripts committed:
Merge branch 'master' into updatesettle-fixpedbugs

MonkeScripts committed:
Merge pull request #166 from avrilgk/updatesettle-fixpedbugs

Updated settle and fix ped bugs relating to group

Cohii2 committed:
Change Balance printing

Makes it clearer which user owes which user.

Commit:
Edit sequence diagrams

Commit:
Add class diagram

Commit:
fix typo

mukund1403 committed:
Merge pull request #162 from hafizuddin-a/hafiz-DG2.1

Update DG

mukund1403 committed:
integrate money into expense. fix checkstyle errors

mukund1403 committed:
fix merge conflicts

hafizuddin-a committed:
Merge pull request #169 from mukund1403/master

integrate money into expense. fix checkstyle errors

MonkeScripts committed:
Merge pull request #168 from Cohii2/hengJunxiang-Balance

Update Balance Class

Commit:
Fix comments

Cohii2 committed:
Rewrite Balance to work with Money

Cohii2 committed:
Unit Tests for Balance class

Commit:
Remove unnecessary checks

Commit:
Add test cases and other sections

MonkeScripts committed:
Merge pull request #170 from Cohii2/master

Rework Balance Class to work with Money

Cohii2 committed:
Rewrite settle command in Parser

Cohii2 committed:
Implement Settle to work with Money

MonkeScripts committed:
Merge pull request #171 from hafizuddin-a/hafiz-FixDiagrams

Fix diagrams

MonkeScripts committed:
Merge pull request #172 from Cohii2/hengJunxiang-settle

Rewrite Settle

MonkeScripts committed:
Create settle-all function

MonkeScripts committed:
Update Luck and abstract Luck commands

MonkeScripts committed:
Merge branch 'Gambling-with-balance' into slot-money-settle

Cohii2 committed:
Update Docs

Add Architecture Diagram to Developer Guide.
Update Balance docs in User Guide.

Commit:
Print cats and fix checkstyle

Commit:
Fix currency test

mukund1403 committed:
update user guide and Expense class without boolean flag

Cohii2 committed:
Fix Parser Test

hafizuddin-a committed:
Merge pull request #174 from mukund1403/master

update user guide and Expense class without boolean flag

MonkeScripts committed:
Merge pull request #175 from Cohii2/hengJunxiang-Parser

Update Parser Test

Commit:
Update messages

hafizuddin-a committed:
Merge branch 'master' into hafiz-Cats

Cohii2 committed:
Fix Bug

Fix bug where "command /" crashes program

hafizuddin-a committed:
Merge pull request #176 from Cohii2/hengJunxiang-Parser

Fix Bug

hafizuddin-a committed:
Merge pull request #173 from hafizuddin-a/hafiz-Cats

Print cats and fix checkstyle

Cohii2 committed:
Update Printing

Update Printing in Main and settle functions

Commit:
Update UG

mukund1403 committed:
update UG

MonkeScripts committed:
Slot Machine feature

MonkeScripts committed:
Update Help test, remove helper function in Luck

MonkeScripts committed:
Move luck test file location

MonkeScripts committed:
Reorder static variables

MonkeScripts committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T15-3/tp into slot-money-settle

MonkeScripts committed:
Clear checkstyle

MonkeScripts committed:
Fixe checkstyle violation

mukund1403 committed:
fix merge conflicts

mukund1403 committed:
Merge pull request #177 from Cohii2/hengJunxiang-ui

Update Printing

mukund1403 committed:
update print statement

avrilgk committed:
Fixed group and moved to command package

mukund1403 committed:
Merge pull request #181 from avrilgk/Avril-bugfixing

Fixed group and moved to command package

mukund1403 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T15-3/tp

hafizuddin-a committed:
Merge pull request #179 from mukund1403/master

update UG

Commit:
Update PPP

avrilgk committed:
Merge pull request #178 from hafizuddin-a/hafiz-MoreUG

Update UG

mukund1403 committed:
add Javadoc for ExpenseCommand

avrilgk committed:
Merge pull request #182 from mukund1403/master

add Javadoc for ExpenseCommand

MonkeScripts committed:
Add ppp

mukund1403 committed:
fix UG table

hafizuddin-a committed:
Merge pull request #183 from mukund1403/master

fix UG table

avrilgk committed:
Fixed UG regarding Group

avrilgk committed:
Fixed UG regarding Group

MonkeScripts committed:
Merge branch 'master' into slot-money-settle

MonkeScripts committed:
Add ppp

MonkeScripts committed:
Merge remote-tracking branch 'origin/slot-money-settle' into slot-money-settle

avrilgk committed:
Updated Avril's PPP

Commit:
Update UG

Commit:
More stuff

Cohii2 committed:
Merge pull request #180 from MonkeScripts/slot-money-settle

Slot money settle

hafizuddin-a committed:
Merge branch 'master' into hafiz-UGAgain

hafizuddin-a committed:
Merge pull request #184 from avrilgk/UG-update

Fixed UG regarding Group

avrilgk committed:
Merge branch 'master' into hafiz-UGAgain

avrilgk committed:
Merge pull request #185 from hafizuddin-a/hafiz-UGAgain

UG edits

Commit:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T15-3/tp into hafiz-UGAgain

# Conflicts:
#	docs/UserGuide.md
#	src/main/java/seedu/duke/Help.java

Commit:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T15-3/tp into hafiz-UGAgain

# Conflicts:
#	docs/UserGuide.md

avrilgk committed:
Merge pull request #186 from hafizuddin-a/hafiz-UGAgain

Hafiz UG

MonkeScripts committed:
correct UG

MonkeScripts committed:
Update luck test

mukund1403 committed:
add PPP

Commit:
Add image

Commit:
Updater main page

MonkeScripts committed:
remove unmappable char

MonkeScripts committed:
remove unused import

MonkeScripts committed:
remove checkstyle issues

hafizuddin-a committed:
Merge pull request #187 from MonkeScripts/master

correct UG

mukund1403 committed:
Merge pull request #189 from hafizuddin-a/hafiz-UGAgain

Hafiz image

Cohii2 committed:
Update AboutUs

Cohii2 committed:
Add UML Diagrams

Cohii2 committed:
Merge pull request #190 from Cohii2/hengJunxiang-docs

Update Docs

Commit:
Fix DB

MonkeScripts committed:
UML files and images

Cohii2 committed:
Update Docs

Update Implementation of Balance Feature in Developer Guide

MonkeScripts committed:
updated UML DG

Cohii2 committed:
Merge pull request #193 from Cohii2/hengJunxiang-docs

Update Docs

Cohii2 committed:
Merge pull request #194 from MonkeScripts/master

Sequence diagrams

mukund1403 committed:
Merge pull request #188 from mukund1403/master

add PPP

mukund1403 committed:
Merge pull request #192 from avrilgk/avril-ppp-update

Updated Avril's PPP

Commit:
Fix check

mukund1403 committed:
Update AboutUs.md

Cohii2 committed:
Update Docs

Cohii2 committed:
Merge pull request #195 from Cohii2/hengJunxiang-docs

Update Docs

mukund1403 committed:
update dg

MonkeScripts committed:
updated Luck structure diagram

Commit:
Update no special char

MonkeScripts committed:
update luck structure image

avrilgk committed:
update docs

avrilgk committed:
Merge branch 'master' into UML-diagrams

Cohii2 committed:
Fix Formatting for DG

avrilgk committed:
Merge pull request #196 from avrilgk/UML-diagrams

update docs

mukund1403 committed:
update dg

Cohii2 committed:
Merge pull request #197 from Cohii2/hengJunxiang-docs

Fix Formatting for DG

Commit:
Add range of expense

MonkeScripts committed:
fix formatting image issue

Cohii2 committed:
Change Location of Balance Structure Diagram

Cohii2 committed:
Merge branch 'master' of https://github.com/AY2324S2-CS2113-T15-3/tp into hengJunxiang-docs

Cohii2 committed:
Fix Error

hafizuddin-a committed:
Merge pull request #191 from hafizuddin-a/hafiz-UGAgain

Fix DB

MonkeScripts committed:
Merge pull request #199 from MonkeScripts/master

Added Luck Structure diagram

Cohii2 committed:
Fix Checkstyle

MonkeScripts committed:
Merge pull request #198 from mukund1403/master

Update dg for expense

MonkeScripts committed:
Merge pull request #200 from Cohii2/hengJunxiang-docs

Change Location of Balance Structure Diagram

MonkeScripts committed:
updated PPP

MonkeScripts committed:
Merge pull request #201 from MonkeScripts/master

updated PPP

mukund1403 committed:
Update mukund1403.md

Cohii2 committed:
Fix Error

mukund1403 committed:
Merge pull request #202 from Cohii2/master

Fix Error
